src/extensions/default/CodeFolding/main.js

```diff
@@ -284,10 +268,118 @@ define(function (require, exports, module) {
     }
 
     /**
+      * Utility function to check if the code folding menu options exist in the menu
+      */
+    function menuExists(id) {
+        var viewMenu = Menus.getMenu(Menus.AppMenuBar.VIEW_MENU);
+        return Menus.getMenuItem(viewMenu._getMenuItemId(id));
+    }
+
+    /**
+      * Create the codefolding menu items and register key bindings if they dont already exist
+      */
+    function createMenuItems() {
+        //register commands
+        if (!CommandManager.get(COLLAPSE_ALL)) {
+            CommandManager.register(Strings.COLLAPSE_ALL, COLLAPSE_ALL, collapseAll);
+        }
+
+        if (!CommandManager.get(EXPAND_ALL)) {
+            CommandManager.register(Strings.EXPAND_ALL, EXPAND_ALL, expandAll);
+        }
+
+        if (!CommandManager.get(COLLAPSE)) {
+            CommandManager.register(Strings.COLLAPSE_CURRENT, COLLAPSE, collapseCurrent);
+        }
+
+        if (!CommandManager.get(EXPAND)) {
+            CommandManager.register(Strings.EXPAND_CURRENT, EXPAND, expandCurrent);
+        }
+
+        //create menus
+        if (!menuExists(codeFoldingMenuDivider)) {
```

Comment:
I'm not clear on why we need to check all these individually.  If one of them exists, they all should...  But also, it seems like we could just maintain an internal flag like `isInitialized` to prevent doing _everything_ in `init()` when it's already been done.  (I'm assuming the core problem here is that preferences broadcasts change events sometimes when the pref value hasn't actually changed, which could lead to trying to initialize twice and thus duplicating all the menu items).